### PR TITLE
sd-netlink: routing policy rule port to fib_rule_hdr

### PR DESCRIPTION
**Pull request by ssahani**

### CODE REVIEW

File: src/libsystemd/sd-netlink/rtnl-message.c [hunk 1]
```diff
@@ -861,177 +862,175 @@ int sd_rtnl_message_new_routing_policy_rule(sd_netlink *rtnl, sd_netlink_message
...
-        rtm->rtm_scope = RT_SCOPE_UNIVERSE;
-        rtm->rtm_type = RTN_UNICAST;
+        frh = NLMSG_DATA((*ret)->hdr);
+        frh->family = ifal_family;
+        frh->action = FR_ACT_TO_TBL;
 
         return 0;
 }
 
 int sd_rtnl_message_routing_policy_rule_set_tos(sd_netlink_message *m, unsigned char tos) {
```
Comment (yuwata) [inline src/libsystemd/sd-netlink/rtnl-message.c]: Please adjust the type: `unsigned char` -> `uint8_t`

File: src/libsystemd/sd-netlink/rtnl-message.c [hunk 2]
```diff
@@ -861,177 +862,175 @@ int sd_rtnl_message_new_routing_policy_rule(sd_netlink *rtnl, sd_netlink_message
...
-        routing_policy_rule = NLMSG_DATA(m->hdr);
+        frh = NLMSG_DATA(m->hdr);
 
-        routing_policy_rule->rtm_tos = tos;
+        frh->tos = tos;
 
         return 0;
 }
 
 int sd_rtnl_message_routing_policy_rule_get_tos(const sd_netlink_message *m, unsigned char *tos) {
```
Comment (yuwata) [inline src/libsystemd/sd-netlink/rtnl-message.c]: Same here.

**Review by yuwata — CHANGES_REQUESTED** [review]
Thanks. Mostly OK. But please adjust the type.

**Comment by ssahani** [comment]
Updated thanks for the review.

**Comment by yuwata** [comment]
LGTM.

**Pull request merged**